engine/src/types.rs

```diff
@@ -206,6 +310,18 @@ impl<'a> From<&'a RhsValue> for LhsValue<'a> {
             RhsValue::Bytes(bytes) => LhsValue::Bytes(Cow::Borrowed(bytes)),
             RhsValue::Int(integer) => LhsValue::Int(*integer),
             RhsValue::Bool(b) => match *b {},
+            RhsValue::Map(m) => match *m {},
+        }
+    }
+}
+
+impl<'a> TryFrom<Type> for LhsValue<'a> {
```

Comment:
Conversion from "type" to "LHS value" doesn't seem semantically correct.

E.g. it's not particularly obvious from usage that `LhsValue::try_from(Type::Bool)` would create a map internally.

Also, when a conversion is not fallible, better to use regular `From`.

Reply:
The goal of this function is to create default `LhsValue` for `Type` that supports it.
`LhsValue::try_from(Type::Bool)` would return an error because we do not handle default value for `Type::Bool` however `LhsValue::try_from(Type::Map(Box::new(Type::Bool)))` would work and return an empty map of boolean.

If the `TryForm` trait is not the right thing to implement, how do you think such a function should be called?

Reply:
What about defining an impl on the enum such as ToLhsValue?

I agree with `From`, but in this case conversion can fail?

Reply:
I replaced it with a `to_lhs_value` for the `Type` enum as requested :+1: